src/compiler/scala/tools/nsc/ast/DocComments.scala

```diff
@@ -60,16 +60,18 @@ trait DocComments { self: Global =>
   private def allInheritedOverriddenSymbols(sym: Symbol): List[Symbol] = {
     val getter: Symbol = sym.getter
     val symOrGetter = getter.orElse(sym)
-    if (!symOrGetter.owner.isClass) Nil
-    else symOrGetter.owner.ancestors map (symOrGetter overriddenSymbol _) filter (_ != NoSymbol)
+    if (symOrGetter.owner.isClass)
+      symOrGetter.owner.ancestors.map { ancestor =>
+        symOrGetter.matchingSymbol(ancestor, symOrGetter.owner.thisType)
```

Comment:
could have kept one-liner with placeholder syntax

Reply:
No, the one-liner is quite long. I may have tried it that day. Its lengthiness looks familiar.